### PR TITLE
Clean up hub

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR removes all uses of the old utils in Transformers to migrate to the new version using `cached_file`, and remove said utils from the Transformers library.
It is slightly breaking since we remove objects (in particular `cached_path` is in the main init albeit not documented), but those are all internal tools, so it's okay in my opinion. Only the research examples are left as before.

### CODE REVIEW

File: src/transformers/modeling_tf_utils.py [hunk 1]
```diff
@@ -2156,7 +2156,7 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
         use_auth_token = kwargs.pop("use_auth_token", None)
         revision = kwargs.pop("revision", None)
         trust_remote_code = kwargs.pop("trust_remote_code", None)
-        mirror = kwargs.pop("mirror", None)
+        _ = kwargs.pop("mirror", None)
         load_weight_prefix = kwargs.pop("load_weight_prefix", None)
```
Comment (LysandreJik) [inline src/transformers/modeling_tf_utils.py:2160]: IIRC, the `mirror` kwarg was already not working as expected, right?

**Review by LysandreJik — APPROVED** [review]
Yes! As seen offline, I agree that these are internal utilities and that all of this should be managed through `huggingface_hub` anyway so happy to see this cleanup taking shape!

File: src/transformers/modeling_tf_utils.py (hunk 1 above)
Comment (sgugger) [reply src/transformers/modeling_tf_utils.py:2160]: Yes, it didn't do anything so this doesn't change anything.

**Comment by adib-enc** [comment]
I still had an error mentioning `ImportError: cannot import name 'cached_path' from 'transformers.utils' `
@ `tf.__version__ == '2.11.0'`. What should i do?

**Comment by sgugger** [comment]
You should stop using that function, as it has been removed from `transformers.utils` in the most recent versions.

**Pull request merged**